File: app/src/main/java/com/wenitech/cashdaily/Util/Dialog/TimePickerFragment.java
package com.wenitech.cashdaily.Util.Dialog;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class TimePickerFragment extends DialogFragment{

    private TimePickerDialog.OnTimeSetListener listener;

    public static TimePickerFragment newInstance(TimePickerDialog.OnTimeSetListener listener) {
        TimePickerFragment fragment = new TimePickerFragment();
        fragment.setListener(listener);
        return fragment;
    }

    public void setListener(TimePickerDialog.OnTimeSetListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        final Calendar c = Calendar.getInstance();
        int hora = c.get(Calendar.HOUR_OF_DAY);
        int minuto = c.get(Calendar.MINUTE);

        // Create a new instance of DatePickerDialog and return it
        return new TimePickerDialog(getActivity(), listener, hora, minuto, true);
    }
}

File: app/src/main/java/com/wenitech/cashdaily/commons/Dialog/TimePickerFragment.kt
package com.wenitech.cashdaily.commons.Dialog

import android.app.Dialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.app.TimePickerDialog
import androidx.fragment.app.DialogFragment
import java.util.*

class TimePickerFragment : DialogFragment() {
    private var listener: OnTimeSetListener? = null

    fun setListener(listener: OnTimeSetListener?) {
        this.listener = listener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val c = Calendar.getInstance()
        val hora = c[Calendar.HOUR_OF_DAY]
        val minuto = c[Calendar.MINUTE]

        // Create a new instance of DatePickerDialog and return it
        return TimePickerDialog(activity, listener, hora, minuto, true)
    }

    companion object {
        fun newInstance(listener: OnTimeSetListener?): TimePickerFragment {
            val fragment = TimePickerFragment()
            fragment.setListener(listener)
            return fragment
        }
    }
}